src/cluster.c

```diff
@@ -4889,16 +5023,14 @@ int verifyClusterConfigWithData(void) {
          * assigned to this slot. Fix this condition. */
 
         update_config++;
-        /* Case A: slot is unassigned. Take responsibility for it. */
+        /* slot is unassigned. Take responsibility for it. */
         if (server.cluster->slots[j] == NULL) {
             serverLog(LL_WARNING, "I have keys for unassigned slot %d. "
                                     "Taking responsibility for it.",j);
             clusterAddSlot(myself,j);
-        } else {
+        } else if (server.cluster->importing_slots_from[j] != server.cluster->slots[j]) {
             serverLog(LL_WARNING, "I have keys for slot %d, but the slot is "
-                                    "assigned to another node. "
-                                    "Setting it to importing state.",j);
-            server.cluster->importing_slots_from[j] = server.cluster->slots[j];
+                                    "assigned to another node. ", j);
```

Comment:
Trailing space within the warning.

```suggestion
                                    "assigned to another node.", j);
```

Reply:
done